api/test/h3-data/h3-data.spec.ts

```diff
@@ -56,9 +57,49 @@ describe('H3-Data Module (e2e)', () => {
       await createFakeH3Data(fakeTable, fakeColumn);
 
       const response = await request(app.getHttpServer())
-        .get(`/api/v1/h3-data/${fakeTable}/${fakeColumn}`)
+        .get(`/api/v1/h3/data/${fakeTable}/${fakeColumn}`)
         .expect(HttpStatus.OK);
       expect(response.body).toEqual({ data: { '861203a4fffffff': 1000 } });
     });
+    test('When I query a H3 data with a non available resolution, then I should get a proper error message', async () => {
+      const response = await request(app.getHttpServer()).get(
+        `/api/v1/h3/data/doYouLikeMyFakeId?resolution=0`,
+      );
+      expect(response.body.errors[0].meta.rawError.response.message[0]).toEqual(
+        'Available resolutions: 1 to 6',
+      );
+    });
+    test('When I query a H3 data with a resolution value that is not a number, then I should get a proper error message', async () => {
+      const response = await request(app.getHttpServer()).get(
+        `/api/v1/h3/data/doYouLikeMyFakeId?resolution=definitelyNotANumber`,
+      );
+      expect(response.body.errors[0].meta.rawError.response.message[1]).toEqual(
+        'resolution must be a number conforming to the specified constraints',
+      );
+    });
+    test('When I query a H3 data by its ID and it does not exist, then I should get a proper error message', async () => {
+      const FAKE_UUID = '959dc56e-a782-441a-be36-1aaa617ed843';
+      const response = await request(app.getHttpServer()).get(
+        `/api/v1/h3/data/${FAKE_UUID}`,
+      );
+      expect(response.body.errors[0].title).toEqual(
+        `Requested H3 with ID: ${FAKE_UUID} could not been found`,
+      );
+    });
+    test('When I query H3 data with no resolution provided, then I should get I should get each available h3indexes (max resolution)', async () => {
+      const id = await createFakeH3Data(fakeTable, fakeColumn, true);
+      const response = await request(app.getHttpServer()).get(
+        `/api/v1/h3/data/${id}`,
+      );
+      expect(Object.keys(response.body).length).toEqual(400);
```

Comment:
See my comment below about "coloring" the sample data. Once you do that, you can do some logic validation of those 400 values. For example, you can count the number of values that are 0 and assert that. Go from there and add some logic validation, to ensure you are not returning 'potato' 400 times

Reply:
The point of just counting the rows its because the magic here is not about the **values**, since these are persisted by an external dataset. 
The magic here is calculating the parent h3indexes, which are the keys in the retrieved data, that's why I didn't care about the values here. And since this use cases brings max resolution, this should return the full fixture

Thoughts?
PS: Anyways I see value in your suggestion of matching the results for the above case of calculating by resolution